### ISSUE TITLE
[Feature Request] Support gradient checkpointing in FlexFlow

### ISSUE THREAD

**Issue by jiazhihao** [flexflow/FlexFlow#48]
This feature request tracks the progress of supporting gradient checkpointing in FlexFlow. As a reference, the user API can be similar to PyTorch checkpointing: https://pytorch.org/docs/stable/checkpoint.html.

**Comment by lockshaw** [comment]
@jiazhihao Should I keep this issue open?
